app/javascript/packs/form-steps-wait.jsx

```diff
@@ -0,0 +1,143 @@
+import { render } from 'react-dom';
+import { Alert } from '@18f/identity-components';
+import { loadPolyfills } from '@18f/identity-polyfill';
+
+/**
+ * @typedef FormStepsWaitElements
+ *
+ * @prop {HTMLFormElement} form
+ */
+
+/**
+ * @typedef FormStepsWaitOptions
+ *
+ * @prop {number} pollIntervalMs Poll interval.
+ * @prop {string} waitStepPath URL path to wait step, used in polling.
+ * @prop {string=} errorMessage Message to show on unhandled server error.
+ */
+
+/** @type {FormStepsWaitOptions} */
+const DEFAULT_OPTIONS = {
+  pollIntervalMs: 3000,
+  waitStepPath: `${window.location.pathname}_wait`,
+};
+
+/**
+ * Returns trimmed content from HTML for given DOM selector, or null if the element does not exist.
+ *
+ * @param {string} html HTML markup.
+ * @param {string} selector DOM selector to retrieve.
+ *
+ * @return {string?} Trimmed content, if exists.
+ */
+export function getContentFromHTML(html, selector) {
+  const dom = document.implementation.createHTMLDocument();
+  dom.body.innerHTML = html;
+  const textContent = dom.querySelector(selector)?.textContent;
+  return textContent ? textContent.trim() : null;
```

Comment:
What about the null-safe navigator? Or is that different because it returns `undefined` instead of `null`?

```suggestion
  return textContent?.trim();
```

Reply:
> What about the null-safe navigator? Or is that different because it returns `undefined` instead of `null`?

Yeah, your suggestion can work, but as you mention, it returns `undefined`. Purely for the sake of keeping the return type to `string` or `null`, the ternary is used.

It could be simplified even further if we'd not mind a return type of `string?=`:

```js
return dom.querySelector(selector)?.textContent?.trim();
```

Reply:
> It could be simplified even further if we'd not mind a return type of `string?=`:

That works for me!